### ISSUE TITLE
update whitelist to get additional url parameter from Nightly bug report

### ISSUE THREAD

**Issue by cch5ng** [webcompat/webcompat.com#1662]
related to https://bugzilla.mozilla.org/show_bug.cgi?id=1381545

after the bug above is merged, a webcompat.com bug reported from the FF Nightly menu button is  expected to include additional URL parameters.

if stylo is enabled, this string will be appended to the webcompat.com bug form url:
"&details=layout.css.servo.enabled: true&label=type-stylo" (plus relevant url escape codes)

- [x] update whitelist
- [x] verify label data is getting sent to github api
- [x] fix text area height
- [x] update functional tests
- [x] add github label for prod issues repo ("type-stylo")

?may want to add a couple functional tests
X - one similar to cur functional test for media report tool (url parameters should populate fields)
* negative case for url param (url without servo pref should leave text area as is)
* one to check the label add for new issue? (not very sure how this should be working)
  * create issue with url param and new issue should have label associated?

**Comment by miketaylr** [comment]
We want to add "type-stylo" to https://github.com/webcompat/webcompat.com/blob/master/config/__init__.py#L65.

**Comment by miketaylr** [comment]
(removing add-on label to not get confusing, this issue is about the built-in menu item in Firefox. Our add-ons live over here: https://github.com/webcompat/webcompat-reporter-extensions)

**Comment by zoepage** [comment]
@miketaylr as this issue is related to Firefox, could we provide an alternative?

**Comment by miketaylr** [comment]
> @miketaylr as this issue is related to Firefox, could we provide an alternative?

An alternative label? I guess something that relates to the form. We probably don't have to spend too much time thinking about it -- @cch5ng should have it fixed this week. :)

**Comment by cch5ng** [comment]
@miketaylr I made the config update as suggested. I also added a new label: 'type-stylo' to the repo:
https://github.com/webcompat/webcompat-tests/labels

however, when I'm trying to test end-to-end from my local Nightly build to saving a new issue, the issue now gets saved but the expected label does not appear in the description. I guess this requires an update to the BE python?

I guess I will hold this pr until after the bugzilla issue lands...

**Comment by miketaylr** [comment]
> I'm not clear of the data source for the call, form_object.get('label', None)

If you're passing in a `&label=type-stylo` GET param in the URL when the form is opened, it gets saved to the session object:

https://github.com/webcompat/webcompat.com/blob/2fa22118e5c2b23c552214c127d99d002e1c6353/webcompat/views.py#L165-L172

Then, when you POST to the new issue endpoing, that is pulled out of the session and attached to the `form` object (which gets passed to `report_issue`):

https://github.com/webcompat/webcompat.com/blob/2fa22118e5c2b23c552214c127d99d002e1c6353/webcompat/views.py#L201

So if that label was added to `EXTRA_LABELS`, it will get passed to GitHub in its POST data:

https://github.com/webcompat/webcompat.com/blob/2fa22118e5c2b23c552214c127d99d002e1c6353/webcompat/form.py#L255
https://github.com/webcompat/webcompat.com/blob/93662e358169e7443445217fe7d2a5c6151c9b21/webcompat/issues.py#L23-L26

If you're not seeing that `type-servo` in the POST data to GitHub, there's a bug somewhere in the chain.

**Comment by cch5ng** [comment]
I believe the webcompat.com text area height will need to be increased. when the pref: layout.css.servo.enabled = true, the string gets added to the text area as expected but at the current height, it does not display by default (and no scrollbar displays to indicate additional content)

<img width="549" alt="screen shot 2017-07-20 at 4 53 40 pm" src="https://user-images.githubusercontent.com/4755432/28443634-3f8b9af0-6d6c-11e7-9dff-170654d350ce.png">

only after you select and manually scroll to the bottom of the field does the string appear

<img width="547" alt="screen shot 2017-07-20 at 4 53 50 pm" src="https://user-images.githubusercontent.com/4755432/28443651-50a495a8-6d6c-11e7-9a2f-f12c75c67ec1.png">

**Comment by cch5ng** [comment]
@miketaylr 
> If you're not seeing that type-stylo in the POST data to GitHub, there's a bug somewhere in the chain.

on investigation, it looks like from the Nightly > Report Site Issue button, the url format looks like:
http://localhost:5000/?url=https%3A%2F%2Fcarrotech.wordpress.com%2F&src=desktop-reporter&details=layout.css.servo.enabled%3A+true&label=type-stylo

but it looks like the webcompat python code is expecting the url to include the "suffix": "/issues/new" to trigger the logic that results in sending the label data to the github api post request

supporting:
https://github.com/webcompat/webcompat.com/blob/master/webcompat/views.py#L165-L172
https://github.com/webcompat/webcompat.com/blob/master/webcompat/form.py#L254-L256

 if I manually update the Report Site Issue button's issue url to
http://localhost:5000/issues/new?url=https%3A%2F%2Fcarrotech.wordpress.com%2F&src=desktop-reporter&details=layout.css.servo.enabled%3A+true&label=type-stylo
... and save that issue, then the expected label displays for the new issue.

do you want me to update the url on the Nightly side or should the logic be updated on the webcompat.com side?

**Comment by miketaylr** [comment]
@cch5ng 

> but it looks like the webcompat python code is expecting the url to include the "suffix": "/issues/new" to trigger the logic that results in sending the label data to the github api post request

Yes, and that's where Firefox will send users by default:

http://searchfox.org/mozilla-central/source/modules/libpref/init/all.js#4939

> do you want me to update the url on the Nightly side or should the logic be updated on the webcompat.com side?

I don't think anything needs to be updated. What is the value of `extensions.webcompat-reporter.newIssueEndpoint` in your local setup?

**Comment by cch5ng** [comment]
just in case: issue with sending label data to github api was resolved by changing Nightly preferences:

about:config?filter=newIssueEndpoint
should be "localhost://5000/issues/new"

(ie the problem was caused by setting the endpoint to "localhost://5000")

**Comment by cch5ng** [comment]
@miketaylr I ended up creating a checklist of todo's for this issue (it grew)
https://github.com/webcompat/webcompat.com/issues/1662#issue-243554112

so far I just created one new functional test to check if the new issue url contains the servo pref, a new string should be detected from the text area; also added skip() functions for 3 old media type report related tests (instead of removing them)

was wondering if I should add a test for creating a new issue with the type-stylo label but couldn't see a similar test for creating an issue and verifying it gets saved (did I just miss it somewhere? or do you avoid those types of tests for some reason? too much time to get the results?)...

**Comment by miketaylr** [comment]
> so far I just created one new functional test to check if the new issue url contains the servo pref, a new string should be detected from the text area; also added skip() functions for 3 old media type report related tests (instead of removing them)

cool. 


> was wondering if I should add a test for creating a new issue with the type-stylo label but couldn't see a similar test for creating an issue and verifying it gets saved (did I just miss it somewhere? or do you avoid those types of tests for some reason? too much time to get the results?)...

no, for now we don't create issues to avoid github labeling us as spammers.

> add github label for prod issues repo ("type-stylo")

(this already exists)
